frontend/src/lib/k8s/lease.ts

```diff
@@ -14,6 +14,22 @@ export interface KubeLease extends KubeObjectInterface {
 
 export class Lease extends makeKubeObject<KubeLease>('Lease') {
   static apiEndpoint = apiFactoryWithNamespace('coordination.k8s.io', 'v1', 'leases');
+  static kind = 'Lease';
```

Comment:
I don't think we should repeat the kind here. The kind should correspond to the objectName we pass to it. There are a couple of places where this is not the case, but we can rectify those. In those cases where it doesn't match, please use the kind hardcoded in the getBaseObject instead of creating a new static var.

Reply:
Removed the static vars and introduced a new parameter for `getBaseObject()` in cluster.ts to take the kind from